modules/core/vault-helper.sh

```diff
@@ -47,9 +47,37 @@ function assert_is_installed {
   fi
 }
 
+function get_required_terraform_data_source_output {
+  local readonly data_source_name="$1"
+  local readonly output_name="$2"
+  local output_value
+
+  output_value=$(terragrunt show -json | jq -r "
+    .values.root_module.resources |
+    .[] |
+    select(.address==\"data.terraform_remote_state.$data_source_name\") |
+    .values.outputs.$output_name")
+
+  if [[ "$output_value" == "null" ]]; then
+    log_error "Unable to find Terraform data source $data_source_name with output $output_name"
+    exit 1
+  fi
+
+  echo "$output_value"
+}
+
 function get_optional_terraform_output {
   local readonly output_name="$1"
-  "${TERRAFORM}" output -no-color "$output_name"
```

Comment:
Does the original not work?

Reply:
Yes it no longer works, if the value is optional and is indeed empty, it will return `null` instead of the previously expected `` empty value. This is probably due to the type changes since TF 0.12, where `null` is more officially recognized as a value lol.

Hence the extra code to check for `null`. The recommendation is to get a JSON output instead of a layman value: https://www.terraform.io/docs/cli/commands/output.html#use-in-automation